quickwit-config/src/config.rs

```diff
@@ -208,8 +193,88 @@ pub struct QuickwitConfig {
     #[serde(rename = "searcher")]
     #[serde(default)]
     pub searcher_config: SearcherConfig,
-    #[serde(rename = "storage")]
-    pub storage_config: Option<StorageConfig>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
This is never meant to be deserialized, so why do we derive  `Deserialize` and some attributes have some serde annotations?